Services/LaunchServer/Launcher.h

```diff
@@ -27,26 +27,38 @@
 #pragma once
 
 #include <AK/HashMap.h>
+#include <AK/HashTable.h>
 #include <AK/URL.h>
 #include <LibCore/ConfigFile.h>
 
 namespace LaunchServer {
 
+struct Handler {
+    String name;
+    String executable;
+    HashTable<String> file_types;
+    HashTable<String> protocols;
+};
+
 class Launcher {
 public:
     Launcher();
     static Launcher& the();
 
+    void load_handlers(const String& af_dir);
     void load_config(const Core::ConfigFile&);
-    bool open_url(const URL&);
+    bool open_url(const URL&, const String handler_name);
     Vector<String> handlers_for_url(const URL&);
 
 private:
+    HashMap<String, Handler> m_handlers;
     HashMap<String, String> m_protocol_handlers;
     HashMap<String, String> m_file_handlers;
 
+    Vector<String> handlers_for(const String& key, HashMap<String, String> user_preferences, Function<bool(Handler, const String&)> handler_matches);
```

Comment:
Please pass parameters by reference to avoid deep copies:
```suggestion
    Vector<String> handlers_for(const String& key, const HashMap<String, String>& user_preferences, Function<bool(const Handler&, const String&)> handler_matches);
 ```

Reply:
I sprinkled some & around, I think in the right places. If you have a good reference on what's the standard such things that'd be great for a C++ noob like myself.